modules/standard/IO.chpl

```diff
@@ -2169,7 +2169,7 @@ inline proc channel.unlock() {
 proc channel.offset():int(64) {
   var ret:int(64);
   on this.home {
-    this.lock();
+    try! this.lock();
```

Comment:
A future TODO: replace try! this.lock with this.lock; defer this.unlock();
I think it's reasonable to leave this as try! for now, but we need to record a TODO somewhere. In functions that throw, it seems odd to use try!.